auth0/src/main/java/com/auth0/android/authentication/storage/SecureCredentialsManager.kt

```diff
@@ -756,6 +715,29 @@ public class SecureCredentialsManager @VisibleForTesting(otherwise = VisibleForT
         fragmentActivity!!.clear()
     }
 
+    /**
+     * Helper method to stores the given [SessionTransferCredentials] refresh token in the storage.
+     * Method will silently return ,if the passed credentials has no refresh token.
+     *
+     * @param sessionTransferCredentials the credentials to save in the storage.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun saveSessionTransferCredentials(sessionTransferCredentials: SessionTransferCredentials) {
+        val existingCredentials: Credentials = try {
+            getExistingCredentials()
+        } catch (exception: CredentialsManagerException) {
+            Log.e(TAG, "Error while fetching existing credentials", exception)
+            return
+        }
+        // Checking if the existing one needs to be replaced with the new one
+        if (existingCredentials.refreshToken == sessionTransferCredentials.refreshToken && existingCredentials.idToken == sessionTransferCredentials.idToken) return
```

Comment:
There's no need to check the ID token for equality, as the newer one will have a different `iat` (issued at) claim value, so the token values will always be different.

Reply:
So we can always replace the old ID token with the new one.